### PR TITLE
cosmic-settings-daemon: init at unstable-2023-12-29

### PR DESCRIPTION
**Pull request by nyabinary**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/co/cosmic-settings-daemon/package.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  src = fetchFromGitHub {
+    owner = "pop-os";
+    repo = pname;
+    rev = "29b47fed5f64e1f2ee3494d9230256c9fb1873aa";
+    hash = "sha256-+AAy8iVYCCJUVsA5oDRsWu9tYF6x1m4yiebw5CTz4i0=";
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+  };
```
Comment (SuperSandro2000) [inline pkgs/by-name/co/cosmic-settings-daemon/package.nix:22]: I don't think we need the lockfile here and can just use the hash.

File: pkgs/by-name/co/cosmic-settings-daemon/package.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+, pkg-config
+, udev
+
```
Comment (SuperSandro2000) [inline pkgs/by-name/co/cosmic-settings-daemon/package.nix:6]:
```suggestion
```

File: pkgs/by-name/co/cosmic-settings-daemon/package.nix (hunk 1 above)
Comment (nyabinary) [reply pkgs/by-name/co/cosmic-settings-daemon/package.nix:22]: Same reasoning as the other review comment I left (git dependencies are not supported by rustPlatform at this time)

Comment (SuperSandro2000) [reply pkgs/by-name/co/cosmic-settings-daemon/package.nix:21]: but there are also none used here

Comment (nyabinary) [reply pkgs/by-name/co/cosmic-settings-daemon/package.nix]: Fixed & changed

**Review by SuperSandro2000 — APPROVED** [review]

**Review by nbdd0121 — APPROVED** [review]

**Pull request merged**